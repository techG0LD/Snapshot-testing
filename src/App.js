
import './App.css';
import GithubCard from './components/GitHubCard';


function App() {
  return (
    <div className="App">
     <GithubCard/>
    </div>
  );
}

export default App;
