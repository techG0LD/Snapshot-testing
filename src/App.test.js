import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'

test('renders a snapshot of app', () => {
  const tree = renderer.create(<App />).toJSON()
 
  expect(tree).toMatchSnapshot();
});

