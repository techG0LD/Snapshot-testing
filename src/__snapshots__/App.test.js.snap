// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`renders a snapshot of app 1`] = `
<div
  className="App"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="https://eu-images.contentstack.com/v3/assets/blt949ea8e16e463049/blt93445186c28dee8e/63945e7b10ec6026c7ebc2bb/E1RR80.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        techG0LD
      </div>
      <p
        className="card-text"
      >
        Check out my github account!
      </p>
    </div>
    <div
      className="list-group-flush list-group"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        I like building front-end
      </div>
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        I enjoy back-end
      </div>
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        my profession is full-stack
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a snapshot4 1`] = `
<div
  className="App"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="https://eu-images.contentstack.com/v3/assets/blt949ea8e16e463049/blt93445186c28dee8e/63945e7b10ec6026c7ebc2bb/E1RR80.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        techG0LD
      </div>
      <p
        className="card-text"
      >
        Check out my github account!
      </p>
    </div>
    <div
      className="list-group-flush list-group"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        I like building front-end
      </div>
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        I enjoy back-end
      </div>
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        my profession is full-stack
      </div>
    </div>
  </div>
</div>
`;
